llama/control_vectors.cpp

```diff
@@ -0,0 +1,40 @@
+#include "control_vectors.h"
+#include "common.h"
+
+extern "C" {
+
+int32_t llama_apply_control_vector(
+    const struct llama_model * model,
+    struct llama_context * lctx,
+    char* path,
+    float strength
+) {
+    // TODO support multiple control vectors at once
+    struct common_control_vector_load_info info;
+    info.strength = strength;
+    info.fname = std::string(path);
+
+    std::vector<common_control_vector_load_info> infos;
```

Comment:
Currently this function only takes and applies a single control vector `(file path, strength)`. This needs to be modified so that it can take an array of control vectors and apply them all at once. I need to validate the logic in llama.cpp before I add that feature